tests/recipes/test_lora_dpo_single_device.py

```diff
@@ -191,3 +191,49 @@ def test_save_and_load_merged_weights(self, tmpdir, monkeypatch):
         llama2_model.load_state_dict(sd)
         merged_ckpt_out = llama2_model(inputs)
         torch.testing.assert_close(baseline_out, merged_ckpt_out, rtol=1e-5, atol=1e-5)
+
+    @pytest.mark.integration_test
+    def test_lr_scheduler_optional(self, tmpdir, monkeypatch):
```

Comment:
this is a great test but it seems expensive just to check that the lr_scheduler is not created. Maybe you could just call the recipe's `setup()` method with a toy model and check that `self._lr_scheduler == None`?

Reply:
I initially wanted to go with this solution, but I do not think the `recipes` were designed to be imported in tests:

https://github.com/pytorch/torchtune/blob/912af640a422d84ce261c100f804490a5c12e8e7/recipes/__init__.py#L7-L8

Reply:
I initially wanted to go with this approach, but I do not think we are allowed to import anything in `recipes` (i.e. `LoRADPORecipeSingleDevice`) from tests:

https://github.com/pytorch/torchtune/blob/912af640a422d84ce261c100f804490a5c12e8e7/recipes/__init__.py#L19-L23